tests/generative/variable_strategies.py

```diff
@@ -195,12 +204,17 @@ def variable(patient_tables, event_tables, schema, int_values, bool_values):
 # We also record the QM operations for which strategies have been created and then assert that
 # this includes all operations that exist, to act as a reminder to us to add new operations here
 # when they are added to the query model.
-def qm_builds(type_, *arg_strategies):
+def qm_builds(type_, *arg_strategies, avoided_inputs=None):
     included_operations.add(type_)
 
     @st.composite
     def strategy(draw, type_, *arg_strategies):
         args = [draw(s) for s in arg_strategies]
+
+        if avoided_inputs:
+            for input_type in avoided_inputs:
+                hyp.assume(not any(isinstance(arg, input_type) for arg in args))
```

Comment:
I'm a bit uncomfortable about this because, I _think_, it gives up on the attempt to build an `IsNull` if this assumption doesn't hold; so it seems like it might result in under-sampling of node types that have avoided inputs.

I've run this locally and did get a reasonable number of `IsNull` operations. I've also got an alternative implementation locally which keeps calling `draw()` until it gets something it likes instead of bailing out of `qm_builds()`. (Currently testing to see if it results in an obviously larger number of `IsNull`s.)

Reply:
Over 10k examples I see 33 `IsNull` operations with the version here and 47 with the alternative implementation. Maybe that's a meaningful difference.